src/main/java/com/mongodb/jdbc/MongoSQLResultSetMetaData.java

```diff
@@ -50,12 +50,15 @@ private void assertDatasourceSchema(MongoJsonSchema schema) throws SQLException
         }
     }
 
-    private void processDataSource(MongoJsonSchema schema, String datasource) throws SQLException {
+    private void processDataSource(MongoJsonSchema schema, String datasource, boolean isDatabaseMetadataResultSet) throws SQLException {
```

Comment:
```suggestion
    private void processDataSource(MongoJsonSchema schema, String datasource, boolean sortFieldsAlphabetically) throws SQLException {
```
[bike-shedding-nit] Should we give a more general-purpose name to this parameter and use it as a positive indicator instead of a negative one (meaning we could say `if (param)` instead of `if (!param)`).

We're using this parameter to determine whether or not columns should be sorted alphabetically. Although the only use case right now is to prevent result set metadata for `DatabaseMetaData` methods from being sorted, there may be other use cases in the future. For example, maybe we'll eventually allow users to specify column order on their queries.

There may never be a use case like that, so this is probably just name bike-shedding. Up to you if you want to consider this!

Reply:
Sounds like a good idea to me. Will make the change.